smart_open/s3.py

```diff
@@ -644,6 +665,134 @@ def __repr__(self):
         )
 
 
+class SinglepartWriter(io.BufferedIOBase):
+    """Writes bytes to S3 using the single part API.
+
+    Implements the io.BufferedIOBase interface of the standard library.
+
+    This class buffers all of its input in memory until its `close` method is called. Only then will
+    the data be written to S3 and the buffer is released."""
+
+    def __init__(
+            self,
+            bucket,
+            key,
+            session=None,
+            resource_kwargs=None,
+            singlepart_upload_kwargs=None,
```

Comment:
I think the singlepart thing is a bit redundant here, because this is the SinglepartWriter. `upload_kwargs` is enough.